xds/internal/xdsclient/authority.go

```diff
@@ -39,44 +35,60 @@ import (
+type xdsChannelWithConfig struct {
+	xc      *xdsChannel
+	sc      *bootstrap.ServerConfig
+	cleanup func()
 }
 
-// authority wraps all state associated with a single management server. It
-// contains the transport used to communicate with the management server and a
-// cache of resource state for resources requested from the management server.
+// authority provides the functionality required to communicate with a
+// management server corresponding to an authority name specified in the
+// bootstrap configuration.
 //
-// Bootstrap configuration could contain multiple entries in the authorities map
-// that share the same server config (server address and credentials to use). We
-// share the same authority instance amongst these entries, and the reference
-// counting is taken care of by the `clientImpl` type.
+// It holds references to one or more xdsChannels, one for each server
+// configuration in the bootstrap, to allow fallback from a primary management
+// server to a secondary management server. Authorities that contain similar
+// server configuration entries will end up sharing the xdsChannel for that
+// server configuration. The xdsChannels are owned and managed by the xdsClient.
+//
+// It also contains cache of resource state for resources requested from the
+// management server. This cache contains the list of registered watchers and
+// the most recent resource configuration received from the management server.
 type authority struct {
-	serverCfg          *bootstrap.ServerConfig       // Server config for this authority
-	bootstrapCfg       *bootstrap.Config             // Full bootstrap configuration
-	refCount           int                           // Reference count of watches referring to this authority
-	serializer         *grpcsync.CallbackSerializer  // Callback serializer for invoking watch callbacks
-	resourceTypeGetter func(string) xdsresource.Type // ResourceType registry lookup
-	transport          *transport.Transport          // Underlying xDS transport to the management server
-	watchExpiryTimeout time.Duration                 // Resource watch expiry timeout
-	logger             *grpclog.PrefixLogger
+	// The following fields are initialized at creation time and are read-only
+	// afterwards, and therefore don't need to be protected with a mutex.
+	name                      string                       // Name of the authority from bootstrap configuration.
+	watcherCallbackSerializer *grpcsync.CallbackSerializer // Serializer to run watcher callbacks, owned by the xDS client implementation.
+	getChannelForADS          xdsChannelForADS             // Function to get an xdsChannel for ADS, provided by the xDS client implementation.
+	serializer                *grpcsync.CallbackSerializer // Serializer to run call ins from the xDS client, owned by this authority.
+	serializerClose           func()                       // Function to close the above serializer.
```

Comment:
Optional: Should this be called xDSChannelSerializer or something with a prefix?